enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,894 @@
+
+   When the IngressController is first created and the LoadBalancer-type service is not yet provisioned, 
+   these same invalid `eipAllocation` values will prevent the load balancer from being provisioned. 
+   The existing Ingress Operator logic will clearly indicate to the cluster admin that the load balancer failed to
+   provision via the LoadBalancerReady status condition. In addition, the LoadBalancerReady condition will include the CCM 
+   event logs that indicate to the cluster admin that the `eipAllocations` value is invalid (see Support Procedures for examples).
+   Therefore, requiring manual service deletion provides a way for the Ingress Operator to produce a clear signal to the
+   cluster admin that `eipAllocations` is invalid.
+
+   See [Invalid EIPAllocation Annotation Values](#invalid-eipallocation-values) for examples of invalid `eipAllocation` annotation values.
+
+   Reason #2: To mitigate upgrade compatibility issues.
+   Note: Directly configuring the `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` on the LoadBalancer-type service is not
+   supported and never has been. However, this enhancement is designed to prevent cluster disruption upon upgrading with this
+   unsupported configuration.
+
+   If the `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` annotation was previously configured and the cluster is upgraded,
+   the default value [] for EIPAllocations will clear the annotation for existing IngressControllers, which would break cluster ingress.
+   However, requiring the service to be deleted to effectuate a `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations`
+   value different from the current value of the annotation prevents this automatic removal of the service annotation on upgrade.
+
+   Note: Cluster admins upgrading with an unmanaged EIP annotation don't need to delete the service to propagate the `eipAllocation` values, 
+   instead they can follow the [Unmanaged EIPAllocation Annotation Migration Workflow](#unmanaged-eipallocation-annotation-migration-workflow).
+
+   Reason #3: The CCM Doesn't Reconcile NLB `eipAllocation` Updates after creation.
+   The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` annotation updates to NLBs 
+   once it has been created. However, requiring a service deletion will effectuate these changes.
+
+   See [CCM Doesn't Reconcile NLB EIPAllocation Updates](#ccm-doesnt-reconcile-nlb-eipallocation-updates) for more details on why the CCM doesn't reconcile these updates.
+ 
+   #####  Effectuating EIP Updates using automatic method:
+   If the admin wants to detect the change in `eipAllocations` and update the LB service automatically,
+   rather than explicitly deleting the Service after changes, they can add the  `ingress.operator.openshift.io/auto-delete-load-balancer` annotation.
```

Comment:
```suggestion
   rather than explicitly deleting the Service after changes, by adding the  `ingress.operator.openshift.io/auto-delete-load-balancer` annotation.
```